### ISSUE TITLE
forge remappings.txt ignored

### ISSUE THREAD

**Issue by alcibiadeseth** [gakonst/foundry#454]
Entries in `remappings.txt` seem to be ignored by the present version of forge.

**Comment by alcibiadeseth** [comment]
```
ls .
lib            remappings.txt src
```

```
cat remappings.txt
@compound/=lib/compound-protocol/
ds-test/=lib/ds-test/src/
@openzeppelin/=lib/openzeppelin-contracts/
```

```
forge remappings
openzeppelin-contracts/=/../lib/openzeppelin-contracts/
compound-protocol/=/../lib/compound-protocol/
ds-test/=/../lib/ds-test/src/
```

It seems like only detected remappings are used.

**Comment by gakonst** [comment]
The `forge remappings` command outputs the auto-detected remappings, whereas the `remappings.txt` is merged into the auto-detected ones inside `forge test`as you can see [here](https://github.com/gakonst/foundry/blob/764c69cc758f15183ce104ccd6c18d9374b03162/cli/src/cmd/build.rs#L161-L187).

Closing as this is intended behavior

**Comment by alcibiadeseth** [comment]
@gakonst what I am saying is that the autodetected mappings are not presently overridden by `remappings.txt`, shouldn't they be?

**Comment by alcibiadeseth** [comment]
Moreover, I figured out the cause, the project is incorrectly detecting the root. I have a git project with a folder called `contracts`, within which is the foundry project, including `remappings.txt`, with a `pwd` of `contracts/`, the project root is incorrectly detected as `../`, which doesn't contain any solidity code. The `remappings.txt` override works successfully if I specify the full path to `contracts/` as a `--root` argument to the tool.

**Comment by gakonst** [comment]
Yep this is a known bug which will be addressed in #297, the workaround to this in monorepos is `forge test --root .`

**Comment by alcibiadeseth** [comment]
Awesome, that makes sense.
